lib/__tests__/test-stack.ts

```diff
@@ -142,7 +142,8 @@ export class TestStack extends Stack {
       identity: 'aws-cdk-dev',
       keySizeBits: 4_096,
       pubKeyParameterName: `/${this.node.path}/CodeSign.pub`,
-      secretName: this.node.path + '/CodeSign',
+      // I am not able to immediately delete the existing secret, it needs to wait at least 7 days.
```

Comment:
The whole chain of resource replacements is:

1. We need to change the custom resource handler's packaging type from zip to Docker image, but Lambda doesn't allow that; we need to make a new Function, which we do by changing the logical ID.
2. Changing the Custom Resource Service Token is similarly forbidden, so we must also create a new one; this is another logical ID change. 
3. Recreating the custom resource will cause it to create a new key, which will fail because the key with that name (`this.node.path + '/CodeSign'`) already exists; so, we must change the key name.

@moelasmar I would think that step 3 would cause the secret to be deleted as well, because the custom resource should delete it on the `DELETE` event. Why doesn't it?